sqlmesh/core/snapshot/evaluator.py

```diff
@@ -284,9 +295,16 @@ def _get_data_objects(schema: exp.Table) -> t.Set[str]:
             }
 
         snapshots_to_create = []
+        versions_to_create: t.Dict[str, t.List[bool]] = defaultdict(list)
         for snapshot, table_names in snapshots_with_table_names.items():
-            if table_names - existing_objects or (snapshot.is_seed and not snapshot.intervals):
+            missing_versions = table_names - existing_objects
+            if missing_versions or (snapshot.is_seed and not snapshot.intervals):
                 snapshots_to_create.append(snapshot)
+                for version in missing_versions or table_names:
+                    versions_to_create[snapshot.name].append(
+                        table_deployability[snapshot.name + version]
+                    )
+                versions_to_create[snapshot.name].sort(key=lambda x: x)
```

Comment:
Why do we sort?

Reply:
This wa for the test assertions that sometimes failed due to a different order of the create in the case of both deployable and non-deployable versions, so this way it would always be in the order [False, True]

Reply:
Also I think the `key` argument is redundant